etl/steps/data/garden/wb/2024-01-17/world_bank_pip.py

```diff
@@ -0,0 +1,1047 @@
+    col_povertygap = []
+    col_tot_shortfall = []
+    col_watts = []
+    col_poverty_severity = []
+    col_decile_share = []
+    col_decile_thr = []
+
+    for p in povlines:
+        col_headcount.append(f"headcount_{p}")
+        col_headcount_ratio.append(f"headcount_ratio_{p}")
+        col_povertygap.append(f"poverty_gap_index_{p}")
+        col_tot_shortfall.append(f"total_shortfall_{p}")
+        col_watts.append(f"watts_{p}")
+        col_poverty_severity.append(f"poverty_severity_{p}")
+
+    for i in range(1, 11):
+        col_decile_share.append(f"decile{i}_share")
+        if i != 10:
+            col_decile_thr.append(f"decile{i}_thr")
+
+    ############################
+    # Negative values
+    mask = (
+        tb[
+            col_headcount
+            + col_headcount_ratio
+            + col_povertygap
+            + col_tot_shortfall
+            + col_watts
+            + col_poverty_severity
+            + col_decile_share
+            + col_decile_thr
+            + ["mean", "median", "mld", "gini", "polarization"]
+        ]
+        .lt(0)
+        .any(axis=1)
+    )
+    tb_error = tb[mask].reset_index(drop=True).copy()
+
+    if not tb_error.empty:
+        log.fatal(
+            f"""There are {len(tb_error)} observations with negative values! In
+            {tabulate(tb_error[['country', 'year', 'reporting_level', 'welfare_type']], headers = 'keys', tablefmt = TABLEFMT)}"""
+        )
+        # NOTE: Check if we want to delete these observations
+        # tb = tb[~mask].reset_index(drop=True)
+
+    ############################
+    # stacked values not adding up to 100%
+    tb["sum_pct"] = tb[col_stacked_pct].sum(axis=1)
+    mask = (tb["sum_pct"] >= 100.1) | (tb["sum_pct"] <= 99.9)
+    tb_error = tb[mask].reset_index(drop=True).copy()
+
+    if not tb_error.empty:
+        log.warning(
+            f"""{len(tb_error)} observations of stacked values are not adding up to 100% and will be deleted:
+            {tabulate(tb_error[['country', 'year', 'reporting_level', 'welfare_type', 'sum_pct']], headers = 'keys', tablefmt = TABLEFMT, floatfmt=".1f")}"""
+        )
+        tb = tb[~mask].reset_index(drop=True).copy()
+
+    ############################
+    # missing poverty values (headcount, poverty gap, total shortfall)
+    cols_to_check = (
+        col_headcount + col_headcount_ratio + col_povertygap + col_tot_shortfall + col_stacked_n + col_stacked_pct
+    )
+    mask = tb[cols_to_check].isna().any(axis=1)
+    tb_error = tb[mask].reset_index(drop=True).copy()
+
+    if not tb_error.empty:
+        log.warning(
+            f"""There are {len(tb_error)} observations with missing poverty values and will be deleted:
+            {tabulate(tb_error[['country', 'year', 'reporting_level', 'welfare_type'] + col_headcount], headers = 'keys', tablefmt = TABLEFMT)}"""
+        )
+        tb = tb[~mask].reset_index(drop=True)
+
+    ############################
+    # Missing median
```

Comment:
Nit - these functions are pretty cool and could be reused in the future. We could have functions like `check_missing_values(tb, col)` or `check_sum_to_100(tb, col)` in `shared.py` module (and eventually move them to `etl/checks.py` if we want to reuse them elsewhere). No need to do it now, you can do it when you realize you could use it elsewhere.